### PR TITLE
Implement join space and receive notification

### PR DESCRIPTION
**Pull request by hyifeng**
This reverts commit 1793db99eb7c78a7ace3897a43c57f489c97fd16.

### CODE REVIEW

**Comment by 2nthony** [comment]
https://github.com/opensquare-network/bounties/issues/249#issuecomment-1296501998

File: backend/packages/backend/src/services/proposal.service/common.js [hunk 1]
```diff
@@ -86,10 +88,26 @@ async function getProposalSpaceByCid(proposalCid) {
...
+async function createSpaceNotifications(space, notificationType, data) {
+  const members = await getSpaceMembers(space);
+
+  for (const member of members) {
+    const receiver = member.memberPublicKey;
+
+    try {
+      await createNotification(receiver, notificationType, data);
+    } catch (e) {
+      // ignore
```
Comment (wliyongfeng) [inline backend/packages/backend/src/services/proposal.service/common.js:100]: We should log the error.

File: backend/packages/backend/src/scripts/space-notify.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    }
+  }
+
+  return await createNotification(status);
+}
+
+async function startNotify() {
+  const proposalCol = await getProposalCollection();
+  const proposals = await proposalCol.find().toArray();
+
```
Comment (wliyongfeng) [inline backend/packages/backend/src/scripts/space-notify.js:109]: Better to add a todo comment to handle proposals with a limited number.

File: backend/packages/backend/src/scripts/space-notify.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+async function handleProposal(proposal) {
+  const status = getProposalStatus(proposal);
+
+  if (proposal.status === status && status !== ProposalStatus.Active) {
+    return;
+  }
+
+  if (proposal.status === "closeToEnd" && status === ProposalStatus.Active) {
```
Comment (wliyongfeng) [inline backend/packages/backend/src/scripts/space-notify.js:91]: We should extract `closeToEnd` as a constant.

**Pull request merged**